duplication ID

with the update of gregtech 1.16.0.750 there is a duplication id which makes crash the game


with tTime: 6/30/21 11:15 PM
Description: There was a severe problem during mod loading that has caused the game to fail

net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from GregTech Energistics (gregtechenergistics)
Caused by: java.lang.IllegalArgumentException: Tried to reassign id 40 to gregtech.api.cover.CoverDefinition@33e49947 (gregtechenergistics:ae2.interface.lv), but it is already assigned to gregtech.api.cover.CoverDefinition@18f687c0 (gregtech:fluid.regulator.lv)!
	at gregtech.api.util.GTControlledRegistry.register(GTControlledRegistry.java:50)
	at gregtech.api.cover.CoverDefinition.registerCover(CoverDefinition.java:28)
	at com.soliddowant.gregtechenergistics.covers.CoverBehaviors.registerBehavior(CoverBehaviors.java:41)
	at com.soliddowant.gregtechenergistics.covers.CoverBehaviors.registerStockerCover(CoverBehaviors.java:34)
	at com.soliddowant.gregtechenergistics.covers.CoverBehaviors.init(CoverBehaviors.java:19)
	at com.soliddowant.gregtechenergistics.CommonProxy.init(CommonProxy.java:24)
	at com.soliddowant.gregtechenergistics.ClientProxy.init(ClientProxy.java:19)
	at com.soliddowant.gregtechenergistics.GregTechEnergisticsMod.init(GregTechEnergisticsMod.java:30)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:637)
	at sun.reflect.GeneratedMethodAccessor8.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:219)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:136)
	at net.minecraftforge.fml.common.Loader.initializeMods(Loader.java:749)
	at net.minecraftforge.fml.client.FMLClientHandler.finishMinecraftLoading(FMLClientHandler.java:336)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:535)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:378)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)e update of gregtech 1.16.0.750 there is a duplication id which makes crash the game

--- a/build.gradle
+++ b/build.gradle
@@ -36,17 +36,13 @@ minecraft {
 
 repositories {
     maven {
-    	name "GregTech Community Edition Maven"
-        url  "https://dl.bintray.com/gregtech/dev" 
-    }
-    maven {
 		name "ChickenBones Maven"
 		url "http://chickenbones.net/maven/"
 	}
-	maven {
-        name "Modmaven for Applied Energistics 2"
-        url "https://modmaven.dev/"
-    }    
+    maven {
+        name = "CurseForge"
+        setUrl("https://minecraft.curseforge.com/api/maven")
+    }  
     maven {
         name = "JEI Maven"
         setUrl("http://dvs1.progwml6.com/files/maven/")
@@ -55,12 +51,13 @@ repositories {
         name = "JEI Maven fallback"
         setUrl("modmaven.k-4u.nl")
     }
+	maven { url = "https://www.cursemaven.com" }
 }
 
 dependencies {
 	deobfCompile "codechicken:CodeChickenLib:1.12.2-3.2.3.358:universal"
-    deobfCompile "gregtech:gregtech:1.12.2-1.10.8.599"
-    deobfCompile "appeng:appliedenergistics2:rv6-stable-7"
+    deobfCompile "gregtechce:gregtech:1.12.2:1.16.1.753"
+    deobfCompile "curse.maven:applied-energistics-2-223794:2747063"
     deobfCompile "mezz.jei:jei_1.12.2:4.15.0.291"
 }
 

--- a/src/main/java/com/soliddowant/gregtechenergistics/covers/CoverBehaviors.java
+++ b/src/main/java/com/soliddowant/gregtechenergistics/covers/CoverBehaviors.java
@@ -14,7 +14,7 @@ import net.minecraft.util.ResourceLocation;
 import java.util.function.BiFunction;
 
 public class CoverBehaviors {
-	public static final int startingCoverId = 40;
+	public static final int startingCoverId = 50;
 	public static void init() {
 		registerStockerCover(0, "ae2.interface.lv", MetaItems.AE2_STOCKER_LV, GTValues.LV, 1280);
 		registerStockerCover(1, "ae2.interface.mv", MetaItems.AE2_STOCKER_MV, GTValues.MV, 5120);

